hs/src/Reach/Eval.hs

```diff
@@ -115,13 +116,15 @@ compileDApp cns (SLV_Prim (SLPrim_App_Delay at opts part_ios top_formals top_s t
         env0 <- locAt slcpi_at $ env_insert "interact" iov top_env_wps
         return $ (slcpi_who, env0)
   penvs <- M.fromList <$> mapM make_penvp part_ios
+  use_strict <- liftIO . dupeIORef =<< asks (sco_use_strict . e_sco)
```

Comment:
Can you put `dupeIORef` in `Reach.Util` and get rid of all the duplicate definitions of it?